### PR TITLE
Vpnaas: VPN service resource

### PR DESCRIPTION
**Pull request by simonre**
Part 2 of a series that addresses #2

### CODE REVIEW

**Comment by jtopjian** [comment]
> Unless I'm misinterpreting the openlab logs I don't think that this is a problem with my code (?)

That's correct -- you can ignore the results of OpenLab acceptance tests with these VPNaaS PRs.

File: openstack/resource_openstack_vpnaas_service_v2.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"NOT_CREATED"},
+		Target:     []string{"PENDING_CREATE"},
+		Refresh:    waitForServiceCreation(networkingClient, service.ID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      0,
+		MinTimeout: 2 * time.Second,
+	}
+	_, err = stateConf.WaitForState()
```
Comment (jtopjian) [inline openstack/resource_openstack_vpnaas_service_v2.go]:
```go
if err != nil {
  return err
}
```

File: openstack/resource_openstack_vpnaas_service_v2.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+		}
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"PENDING_UPDATE"},
+			Target:     []string{"UPDATED"},
+			Refresh:    waitForServiceUpdate(networkingClient, service.ID),
+			Timeout:    d.Timeout(schema.TimeoutCreate),
+			Delay:      0,
+			MinTimeout: 2 * time.Second,
+		}
+		_, err = stateConf.WaitForState()
```
Comment (jtopjian) [inline openstack/resource_openstack_vpnaas_service_v2.go]:
```go
if err != nil {
  return err
}
```

**Comment by simonre** [comment]
@jtopjian Apart from the test issue this is ready for review again.

**Comment by jtopjian** [comment]
@simonre I have no idea why I misread your notes twice.

Since the problem is only with updating, then I'm OK with omitting update tests. If we can test creation, then that's good enough and will exercise a good amount of code in this resource.

**Comment by simonre** [comment]
@jtopjian I removed the update test and rebased the code.

**Pull request merged**